### PR TITLE
exclude connections flag for aea run

### PR DESCRIPTION
**Pull request by solarw**
## Proposed changes

exclude connections flag for aea run

## Fixes

https://github.com/fetchai/agents-aea/issues/1652

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes and CI passes too
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: aea/cli/run.py [hunk 1]
```diff
@@ -75,26 +76,65 @@
...
+    exclude_connection_ids: List[PublicId],
     env_file: str,
     is_install_deps: bool,
     profiling: int,
     password: str,
 ) -> None:
     """Run the agent."""
+    if connection_ids and exclude_connection_ids:
+        raise click.ClickException(
+            "please use only one of --connections or --exclude-connections. not both"
```
Comment (DavidMinarsch) [inline aea/cli/run.py:101]:
```suggestion
            "Please use only one of --connections or --exclude-connections, not both!"
```

File: aea/cli/run.py [hunk 2]
```diff
@@ -75,26 +76,65 @@
...
+    ctx: Context, exclude_connections: List[PublicId]
+) -> List[PublicId]:
+    """Calculate resulting list of connection ids to run."""
+    agent_config_manager = AgentConfigManager.load(ctx.cwd)
+    not_existing_connections = (
+        set(exclude_connections) - agent_config_manager.agent_config.connections
+    )
+    if not_existing_connections:
+        raise ValueError(
+            f"Connections to exclude: {', '.join(map(str, not_existing_connections))} are not defined in agent configuration"
```
Comment (DavidMinarsch) [inline aea/cli/run.py:126]:
```suggestion
            f"Connections to exclude: {', '.join(map(str, not_existing_connections))} are not defined in agent configuration!"
```

File: aea/cli/run.py [hunk 3]
```diff
@@ -75,26 +76,65 @@
...
+        )
+
+    connection_ids = list(
+        agent_config_manager.agent_config.connections - set(exclude_connections)
+    )
+    if not connection_ids:
+        raise ValueError("No connections left after exclusion!")
```
Comment (DavidMinarsch) [inline aea/cli/run.py:133]: That's not a problem. We allow run without connections. Please remove exception here.

**Review by DavidMinarsch — APPROVED** [review]
LGTM

File: aea/cli/run.py (hunk 3 above)
Comment (solarw) [reply aea/cli/run.py]: removed

**Review by 5A11 — APPROVED** [review]
lgtm

**Pull request merged**